src/applyTransformFeatures.js

```diff
@@ -1,43 +1,93 @@
-function toLowerCase (value) {
-    return value.toLowerCase();
+import valueParser, {unit} from 'postcss-value-parser';
+
+function isHex (node) {
+    if (node.value[0] !== '#') {
+        return false;
+    }
+    const range = node.value.slice(1);
+    return ~[3, 4, 6, 8].indexOf(range.length) && !isNaN(parseInt(range, 16));
 }
 
-function toUpperCase (value) {
-    return value.toUpperCase();
+function toShorthand (hex) {
+    if (
+        hex.length === 7 &&
+        hex[1] === hex[2] &&
+        hex[3] === hex[4] &&
+        hex[5] === hex[6]
+    ) {
+        return '#' + hex[2] + hex[4] + hex[6];
+    }
+    return hex;
 }
 
+function toLonghand (hex) {
+    if (hex.length !== 4) {
+        return hex;
+    }
+
+    const r = hex[1];
+    const g = hex[2];
+    const b = hex[3];
+    return '#' + r + r + g + g + b + b;
+};
+
+const lengths = [
+    'px',
+    'em',
+    'rem',
+    'ex',
+    'ch',
+    'vh',
+    'vw',
+    'cm',
+    'mm',
+    'in',
+    'pt',
+    'pc',
+    'vmin',
+    'vmax',
+];
+
```

Comment:
I just noticed a problem with the `toShorthand` and `toLonghand` functions. There are now a bunch of browsers that support 4 and 8-digit hex codes - https://drafts.csswg.org/css-color/#hex-notation.